src/main/java/gregtech/common/covers/detector/CoverDetectorFluidAdvanced.java

```diff
@@ -0,0 +1,172 @@
+import codechicken.lib.vec.Cuboid6;
+import codechicken.lib.vec.Matrix4;
+import gregtech.api.cover.CoverBehavior;
+import gregtech.api.cover.CoverWithUI;
+import gregtech.api.cover.ICoverable;
+import gregtech.api.gui.GuiTextures;
+import gregtech.api.gui.ModularUI;
+import gregtech.api.gui.widgets.*;
+import gregtech.api.util.GTLog;
+import gregtech.api.util.GTUtility;
+import gregtech.client.renderer.texture.Textures;
+import gregtech.common.covers.filter.FluidFilterContainer;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.*;
+import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.CapabilityFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidTankProperties;
+
+import java.util.regex.Pattern;
+
+public class CoverDetectorFluidAdvanced extends CoverDetectorFluid implements CoverWithUI {
+
+    private int min, max;
+    private final int DEFAULT_MIN = 1000; // 1 Bucket
+    private final int DEFAULT_MAX = 16000; // 16 Buckets
+    protected FluidFilterContainer fluidFilter;
+
+    public CoverDetectorFluidAdvanced(ICoverable coverHolder, EnumFacing attachedSide) {
+        super(coverHolder, attachedSide);
+        this.fluidFilter = new FluidFilterContainer(this);
+        this.min = DEFAULT_MIN;
+        this.max = DEFAULT_MAX;
+    }
+
+    @Override
+    public EnumActionResult onScrewdriverClick(EntityPlayer playerIn, EnumHand hand, CuboidRayTraceResult hitResult) {
+        if (!this.coverHolder.getWorld().isRemote) {
+            openUI((EntityPlayerMP) playerIn);
+        }
+        return EnumActionResult.SUCCESS;
+    }
+
+    @Override
+    public void renderCover(CCRenderState renderState, Matrix4 translation, IVertexOperation[] pipeline, Cuboid6 plateBox, BlockRenderLayer layer) {
+        // todo replace with unique texture
+        Textures.DETECTOR_FLUID.renderSided(attachedSide, plateBox, renderState, pipeline, translation);
+    }
+
+    @Override
+    public ModularUI createUI(EntityPlayer player) {
+        int PADDING = 3;
+        int SIZE = 18;
+
+        WidgetGroup group = new WidgetGroup();
+        group.addWidget(new LabelWidget(10, 8, "cover.advanced_fluid_detector.label"));
+
+        // set min fluid amount
+        group.addWidget(new LabelWidget(10, 5 + (SIZE + PADDING), "cover.advanced_fluid_detector.min"));
+        group.addWidget(new ImageWidget(98 - 4, (SIZE + PADDING), 4 * SIZE, SIZE, GuiTextures.DISPLAY));
+        group.addWidget(new TextFieldWidget2(98, 5 + (SIZE + PADDING), 4 * SIZE, SIZE,
+                this::getMinValue, this::setMinValue)
+                .setMaxLength(10)
+                .setAllowedChars(Pattern.compile(".[0-9]*"))
```

Comment:
This regex pattern should be stored as a `static final` constant, to improve performance. This could potentially go into GTUtility, as it is used across many classes.